### ISSUE TITLE
Implement bookmarks sorting by tags.

### ISSUE THREAD

**Issue by ioctl-user** [organicmaps/organicmaps#4765]
Please consider adding tags to bookmarks. Can be used to select bookmarks for one country, for example.

**Comment by biodranik** [comment]
Selecting bookmarks by country can be implemented natively, without any manual tags. Are there any better use-case examples?

**Comment by ioctl-user** [comment]
It can be used for the different types of the bookmarks of the same region. And some types of the bookmarks may cover few regions

For example: mushroom spots + touristic attractions + camping places.

**Comment by biodranik** [comment]
Why not create separate lists for each of the listed categories in your example, instead of using tags?

**Comment by ioctl-user** [comment]
That's possible. However, the problem is in upkeeping and searching of needed lists in case of big number of them. For example, I have about 40 lists with tend to have more.

Tags fixes the same problem as lists directories and subdirectories, but more flexible.

**Comment by biodranik** [comment]
Will a convenient search in bookmarks and lists names and descriptions help?

**Comment by ioctl-user** [comment]
With very big number of the lists convenient search may not meed all needs.
However, that's much better than nothing and could be enough for the most cases.

It seems, I cannot understand will search solve my problems or not before trying it.

If you going to implement good search and sorting (by alphabet), just close this feature request. In case of suggestions it can be reopened.

**Comment by biodranik** [comment]
It would be great to fix/implement most obvious and critical issues, and then see how it works.

**Comment by melsophos** [comment]
I really like the idea of having something to organize bookmark lists. A good use case is to organize bookmarks in coarse and fine categories: for example, you could have a folder "Food" with lists "Restaurants", "Take away", "Vegan", etc. (or "Tourism", "Accommodation"...).
Tags have more flexibility than folders, so they would be maybe better, but it can also get a bit messier. In any case I don't think that both should be implemented.
(Similar message also in #4766)

**Comment by yasio90** [comment]
First of all, thank you for Organic Maps — it’s an excellent app and a constant companion when I travel.

Together with several other frequent users, we support adding **tags for bookmarks and tracks**.

#### Motivation

Folders are currently the only way to categorize bookmarks (bookmark colors are not always informative, especially for color-blind users). Tags would provide an additional, orthogonal way to store metadata and filter bookmarks.

#### Phase 1: Tag-based filtering and visibility

The most important feature would be:

* ability to **assign one or more tags** to bookmarks (and tracks),
* ability to **filter bookmarks by tags**,
* ability to **toggle visibility of bookmarks by tag**.

This would allow workflows such as:

* hiding places already visited,
* filtering cheap food vs. fancy restaurants,
* filtering by rating (different ratings as different tags),
* showing only specific categories temporarily (if the map is too crowded with bookmarks).

Ideally, this would work globally, but even **tag-based visibility within a single folder** would already be very useful as a first step.

#### Phase 2: Optional color binding to tags

As a follow-up feature, it would be great to allow:

* binding a **color to a tag**,
* automatically changing a bookmark’s color when a color-bound tag is assigned.

This should not be a hard lock:

* users should still be able to manually override the color,
* tags without a color binding should be supported.

Example: all bookmarks tagged as `restaurant` automatically become orange, without the user needing to remember the color scheme.

Ideally, only one color-bound tag could be active per bookmark. If that’s too restrictive, applying the color of the most recently assigned color-bound tag would also work.

Thanks for considering this request, and for the continued development of Organic Maps.

**Comment by biodranik** [comment]
Thanks! Would custom icons (you can select anything) in custom colors help you at least in some use cases?

**Comment by yasio90** [comment]
Yes, this would definitely help color-blind people! Despite not being strongly color-blind, the different shades of blue are not easily discernable to me. This would a big improvement on its own and together with tags would fulfill almost all my wishes with the bookmarks :D

**Comment by Krzysiek-Ka** [comment]
As much as I love Organic Maps, it does have its limitations. Lack of tags is one of them. Another advantage of tags, not mentioned before, is that they would allow assigning multiple tags to one location (while currently it can exist only in a single list). Example use:
Place: Wieliczka Salt Mine
List: Poland
Tags: ‘UNESCO’, ‘Visited’, ‘4/5’

**Comment by biodranik** [comment]
Adding separate "visited" and "rating" fields may also solve the last example.
